src/material/datepicker/datepicker-input.ts

```diff
@@ -123,25 +129,33 @@ export class MatDatepickerInput<D> implements ControlValueAccessor, OnDestroy, V
 
   /** The value of the input. */
   @Input()
-  get value(): D | null { return this._value; }
+  get value(): D | null { return this._value ? this._value.asDate() : null; }
```

Comment:
No need to check if `_value` is truthy, we'll consider it an error if it isn't set up during construction